protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -246,32 +246,23 @@ impl NetworkBehaviour for Behaviour {
         local_addr: &Multiaddr,
         remote_addr: &Multiaddr,
     ) -> Result<THandler<Self>, ConnectionDenied> {
-        match self
-            .outgoing_direct_connection_attempts
-            .remove(&(connection_id, peer))
+        if is_relayed(local_addr) {
+            Ok(Either::Left(handler::relayed::Handler::new(
+                ConnectedPoint::Listener {
+                    local_addr: local_addr.clone(),
+                    send_back_addr: remote_addr.clone(),
+                },
+            ))) // TODO: We could make two `handler::relayed::Handler` here, one inbound one outbound.
+        } else if let Some(&relayed_connection_id) =
+            self.direct_to_relayed_connections.get(&connection_id)
```

Comment:
Can we use this opportunity to move away from `else` please?

```suggestion
        if is_relayed(local_addr) {
            return Ok(Either::Left(handler::relayed::Handler::new(
                ConnectedPoint::Listener {
                    local_addr: local_addr.clone(),
                    send_back_addr: remote_addr.clone(),
                },
            ))) // TODO: We could make two `handler::relayed::Handler` here, one inbound one outbound.
        }
        
        if let Some(&relayed_connection_id) = self.direct_to_relayed_connections.get(&connection_id)
```

Reply:
Not opposed to the change, but why do you want to move away from `else`?

Reply:
I think @thomaseizinger rational here is to follow the pattern of return-early. Just a matter of style which we have been following across the code-base. I would argue consistency is helpful here.

Reply:
Yes, purely a style thing that we prefer here. It makes it easier to read the fn from top to bottom because you can evaluate one condition at a time and don't have to keep in your head, what the inverse of the `if` is whilst you are trying to understand `else`.

The new `let else` feature from Rust plays in the same league. I am excited for when we bump our MSRV to support it :)

Reply:
I've committed this change.